hadoop-common-project/hadoop-common/src/test/java/org/apache/hadoop/io/compress/TestCodecFactory.java

```diff
@@ -137,9 +138,15 @@ private static CompressionCodecFactory setClasses(Class[] classes) {
   
   private static void checkCodec(String msg, 
                                  Class expected, CompressionCodec actual) {
-    assertEquals(msg + " unexpected codec found",
-                 expected.getName(),
-                 actual.getClass().getName());
+    if (expected == null) {
+      assertNull(msg, actual);
+    } else if (actual == null) {
```

Comment:
Is this tested?

Reply:
Yes; assertNull has a detailed builtin message. Here's an example of the new messages:

When a result was expected:
`java.lang.AssertionError: default factory for .gz result was null`

When null was expected:
`java.lang.AssertionError: default factory for .gz expected null, but was:<org.apache.hadoop.io.compress.GzipCodec@3108bc>`